engine/apps/grafana_plugin/helpers/client.py

```diff
@@ -362,12 +362,20 @@ def get_instances(self, query: str, page_size=None):
             cursor = 0
             while cursor is not None:
                 if query:
-                    page_query = query + f"&cursor={cursor}&pageSize={page_size}"
+                    page_query = f"{query}&cursor={cursor}&pageSize={page_size}"
                 else:
                     page_query = f"?cursor={cursor}&pageSize={page_size}"
-                page, _ = self.api_get(page_query)
+                page, call_status = self.api_get(page_query)
```

Comment:
[`api_get`](https://github.com/grafana/oncall/blob/dev/engine/apps/grafana_plugin/helpers/client.py#L103-L104) (which calls [`call_api`](https://github.com/grafana/oncall/blob/dev/engine/apps/grafana_plugin/helpers/client.py#L112)) is calling [`response.raise_for_status()`](https://github.com/grafana/oncall/blob/dev/engine/apps/grafana_plugin/helpers/client.py#L125) which would mean that the HTTP response in these scenarios is HTTP 200, but for some reason the returned data is still missing this key?

tbd once we get more information from the logs

Reply:
asked for more clarification about this endpoint's behaviour in #grafana-com-dev ([thread](https://raintank-corp.slack.com/archives/C0K031RP1/p1723158123932529))